UnityProject/Assets/Scripts/Objects/LaserReflectObject.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Messages.Server;
+using ScriptableObjects.Gun;
+using UnityEngine;
+using Weapons.Projectiles;
+using Weapons.Projectiles.Behaviours;
+
+namespace Objects
+{
+	public class LaserReflectObject : MonoBehaviour, IOnHitDetect
+	{
+		[SerializeField]
+		//Use to check whether a bullet is a laser
+		private LayerMaskData laserData = null;
+
+		public void OnHitDetect(OnHitDetectData data)
+		{
+			//Only reflect lasers
+			if (data.BulletObject.TryGetComponent<Bullet>(out var bullet) == false || bullet.MaskData != laserData) return;
+
+			ShootAtDirection(GetNewDirection(data), data);
+		}
+
+		private Vector2 GetNewDirection(OnHitDetectData data)
+		{
+			var normal = data.BulletShootNormal;
+			var newDirection = data.BulletShootDirection - 2 * (data.BulletShootDirection * normal) * normal;
```

Comment:
I assume this is copying from the code with the tile version of this?

Reply:
Uses a different parameter but yeah

Reply:
Though the one from the tile version use global variables as well